src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs

```diff
@@ -871,7 +563,33 @@ await foreach (var kvp in _source.GetCompareExchangeValuesAsync(actions))
             return result.CompareExchange;
         }
 
-        private async Task<SmugglerProgressBase.Counts> ProcessCountersAsync(SmugglerResult result)
+        private bool SkipCompareExchange(bool isSharded, CompareExchangeKey Key, int index, DatabaseRecord databaseRecord)
+        {
+            if (isSharded)
+            {
+                if (ClusterTransactionCommand.IsAtomicGuardKey(Key.Key, out var docId))
+                {
+                    var id = docId.Split('/')[1];
+                    using (_transactionContextPool.AllocateOperationContext(out TransactionOperationContext context))
+                    {
+                        var docShardIndex = ShardHelper.GetShardIndexforDocument(context, databaseRecord.ShardAllocations, docId);
+                        if (docShardIndex != index)
+                        {
+                            return true;
+                        }
+                    }
+                }
+                else
+                {
+                    if (index != databaseRecord.ShardAllocations.Count - 1)
```

Comment:
Don't do it like this. Instead, have this as a parameter from the caller.

Reply:
done